pkgs/ffigen/test/native_objc_test/bad_override_test.m

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) 2024, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+#import <Foundation/NSObject.h>
+
+@interface BadOverrideGrandparent : NSObject {}
+@end
+@implementation BadOverrideGrandparent
+@end
+
+@interface BadOverrideParent : BadOverrideGrandparent {}
+-(int32_t)methodVsGetter;
+@property (readonly) int32_t methodVsGetter;
+@end
+@implementation BadOverrideParent
+-(int32_t)methodVsGetter { return 1; }
+@end
+
+@interface BadOverrideUncle : BadOverrideGrandparent {}
```

Comment:
I also have [grandparent](https://github.com/dart-lang/native/blob/main/pkgs/jnigen/test/simple_package_test/java/com/github/dart_lang/jnigen/generics/GrandParent.java) in my test code but no uncle and aunt, I should add them 🤣